core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants-add/dot-experiments-configuration-variants-add.component.ts

```diff
@@ -0,0 +1,130 @@
+import { DotExperimentsConfigurationStore } from '@portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.service';
+
+type DotExperimentAddVariant = DotStoreWithSidebar;
+
+interface AddForm {
+    name: FormControl<string>;
+}
+
+@Component({
+    selector: 'dot-experiments-configuration-variants-add',
+    standalone: true,
+    imports: [
+        CommonModule,
+        ReactiveFormsModule,
+
+        DotSidebarHeaderComponent,
+        DotMessagePipeModule,
+        DotFieldValidationMessageModule,
+        DotSidebarDirective,
+        //PrimeNg
+        SidebarModule,
+        ButtonModule,
+        InputTextModule
+    ],
+    templateUrl: './dot-experiments-configuration-variants-add.component.html',
+    styleUrls: ['./dot-experiments-configuration-variants-add.component.scss'],
+    providers: [ComponentStore],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotExperimentsConfigurationVariantsAddComponent implements OnInit {
+    /**
+     * Emit when the sidebar is closed
+     */
+    @Output()
+    closedSidebar = new EventEmitter<void>();
+
+    vm$ = this.componentStore.state$;
+    form: FormGroup<AddForm>;
+
+    private readonly addVariant = this.componentStore.effect(
+        (variant$: Observable<Pick<DotExperiment, 'name'>>) => {
+            return variant$.pipe(
+                tap(() => this.componentStore.patchState({ isSaving: true })),
+                withLatestFrom(this.dotExperimentsConfigurationStore.getExperimentId$),
+                switchMap(([variant, experimentId]) =>
+                    this.dotExperimentsService.addVariant(experimentId, variant).pipe(
+                        tapResponse(
+                            (experiment) => {
+                                this.messageService.add({
+                                    severity: 'info',
+                                    summary: this.dotMessageService.get(
+                                        'experiments.configure.variant.add.confirm-title'
+                                    ),
+                                    detail: this.dotMessageService.get(
+                                        'experiments.configure.variant.add.confirm-message',
+                                        experiment.name
+                                    )
+                                });
+
+                                this.dotExperimentsConfigurationStore.setTrafficProportion(
+                                    experiment.trafficProportion
+                                );
+                                this.closeSidebar();
+                            },
+                            (error: HttpErrorResponse) => throwError(error)
+                        )
+                    )
+                )
+            );
+        }
+    );
+
+    constructor(
+        private readonly componentStore: ComponentStore<DotExperimentAddVariant>,
```

Comment:
You shound not use ComponentStore directly in the component, you create a service and extend.